CoreScriptsRoot/Modules/ChatSelector.lua

```diff
@@ -0,0 +1,203 @@
+local USE_NEW_CHAT = false --and settings():GetFFlag("UseNewChat")
+
+local CoreGuiService = game:GetService("CoreGui")
+local RobloxGui = CoreGuiService:WaitForChild("RobloxGui")
+
+local Util = {}
+do
+	function Util.Signal()
+		local sig = {}
+
+		local mSignaler = Instance.new('BindableEvent')
+
+		local mArgData = nil
+		local mArgDataCount = nil
+
+		function sig:fire(...)
+			mArgData = {...}
+			mArgDataCount = select('#', ...)
+			mSignaler:Fire()
+		end
+
+		function sig:connect(f)
+			if not f then error("connect(nil)", 2) end
+			return mSignaler.Event:connect(function()
+				f(unpack(mArgData, 1, mArgDataCount))
+			end)
+		end
+
+		function sig:wait()
+			mSignaler.Event:wait()
+			assert(mArgData, "Missing arg data, likely due to :TweenSize/Position corrupting threadrefs.")
+			return unpack(mArgData, 1, mArgDataCount)
+		end
+
+		return sig
+	end
+end
+
+
+local moduleApiTable = {}
+do
+	if (USE_NEW_CHAT) then
+		local StarterGui = game:GetService("StarterGui")
```

Comment:
Rather than having GetService for StarterGui in multiple spots lets just put all the GetServices at the top of the file